specs/protocol/revenue-sharing.md

```diff
@@ -0,0 +1,92 @@
+# Revenue Sharing
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+  - [Definitions](#definitions)
+  - [`RevenueSharer` predeploy](#revenuesharer-predeploy)
+    - [Deploying `RevenueSharer`](#deploying-revenuesharer)
+      - [Existing Chains](#existing-chains)
+      - [Chains after version X.Y.Z of OP Stack](#chains-after-version-xyz-of-op-stack)
+    - [Reconfiguring existing predeploys](#reconfiguring-existing-predeploys)
+      - [Existing Chains](#existing-chains-1)
+      - [Chains after version X.Y.Z of OP Stack](#chains-after-version-xyz-of-op-stack-1)
+    - [Execution](#execution)
+  - [Simplified L1 Data Fee Expenditure](#simplified-l1-data-fee-expenditure)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Overview
+
+Revenue sharing is the process by which chains in the superchain contribute a portion of their revenue or profit to a beneficiary such as the Optimism collective. They may do this in return for support with the OP Stack and/or other benefits. 
+
+
+## Definitions
+| Term   | Name         | Definition  |
+| -------|--------------| ----------- |
+| $d$ | L1 Data Fee Revenue    | ETH transferred to the sequencer with L2 transactions to cover estimated L1 Data Fees(see below). Accumulates to `L1FeeVault`.
+| $e$ | L1 Data Fee Expenditure| ETH spent by the batcher on L1 to make transaction data available
+| $b$ | L2 Base Gas Revenue         | Base fee portion of L2 Gas Fee (ETH transferred to the sequencer to cover execution of L2 transactions). Accumulates to `BaseFeeVault`.
+| $q$ | L2 Priority Gas Revenue         | Priority fee portion of L2 Gas Fee (ETH transferred to the sequencer to cover execution of L2 transactions). Accumulates to `SequencerFeeVault`.
+| $g$ | L2 Gas Revenue         | $b + q$
+| $r$ | Sequencer Revenue      | $d + b + q$
+| $p$ | Sequencer Profit       | $r - e$
+| $s$ | Revenue share due to Optimism Collective | $\max(0.15r,0.025p)$
+
+## `RevenueSharer` predeploy
+Revenue sharing is achieved through an L2 [predeploy](./predeploys.md) contract `RevenueSharer` with address
+
+```
+0x4200000000000000000000000000000000000024
+```
+
+### Deploying `RevenueSharer`
+
+
+#### Existing Chains
+The following transactions need to be executed: 
+1. The `RevenueSharer` should be deployed manually to an arbitrary address `REVENUE_SHARER_IMPLEMENTATION_ADDRESS`
+2. For EACH of the three FeeVault contracts ({name} = BASE,L1, SEQUENCER):
+
+    a. A new version should be deployed  with the immutable value `RECIPIENT` set to `0x4200000000000000000000000000000000000024`  in the constructor, and other constructor parameters set to existing values. This contract has an arbitrary address NEW_{name}_FEE_VAULT_ADDRESS.
```

Comment:
We should be careful about specifying `immutable` here because we would like to do an L2 MCP upgrade, ie move to storage from immutables for predeploys

Reply:
[d87d0b6](https://github.com/ethereum-optimism/specs/pull/115/commits/d87d0b6287222a0d2b82442029f5272df3d1d955)